src/targets/gpu/kernels/include/migraphx/kernels/array.hpp

```diff
@@ -218,14 +218,14 @@ struct array
     {
         for(index_int i = 0; i < N; i++)
         {
-            if(not(x[i] < y[i]))
+            if((x[i] >= y[i]))
                 return false;
         }
         return true;
     }
     friend constexpr bool operator>(const array& x, const array& y) { return y < x; }
     friend constexpr bool operator<=(const array& x, const array& y) { return (x < y) or (x == y); }
-    friend constexpr bool operator>=(const array& x, const array& y) { return (y < x) or (x == y); }
+    friend constexpr bool operator>=(const array& x, const array& y) { return not(x < y); }
```

Comment:
I think `(x >= y) == not(x < y)` is not valid for product order since its not totally ordered. 

https://en.cppreference.com/w/cpp/concepts/totally_ordered
https://en.wikipedia.org/wiki/Product_order

Reply:
Reverted back to as it was before.